src/test/shared/sam/debugger/samDebugConfigProvider.test.ts

```diff
@@ -246,30 +286,76 @@ describe('AwsSamDebugConfigurationProvider', async () => {
                     //projectRoot: 'root as in beer'
                 },
             })
-            assert.strictEqual(resolved!.name, name)
+            // TODO: why not respect caller-chosen name?
+            assert.strictEqual(resolved!.name, 'SamLocalDebug')
         })
-        it('target=code', async () => {
-            const debugConfig = {
+
+        it('target=code: javascript', async () => {
+            const appDir = path.join(getProjectDir(), 'integrationTest-samples/js-manifest-in-root/')
+            const folder = getWorkspaceFolder(appDir)
+            const c = {
                 type: AWS_SAM_DEBUG_TYPE,
                 name: 'whats in a name',
                 request: DIRECT_INVOKE_TYPE,
                 invokeTarget: {
                     target: CODE_TARGET_TYPE,
-                    lambdaHandler: 'sick handles',
-                    projectRoot: 'root as in beer',
+                    lambdaHandler: 'my.test.handler',
+                    projectRoot: 'src',
                 },
                 lambda: {
                     runtime: validRuntime,
                 },
             }
-            assert.deepStrictEqual(
-                await debugConfigProvider.resolveDebugConfiguration(undefined, debugConfig),
-                debugConfig
-            )
+            ;(c as any).configOnly = true
+            const actual = (await debugConfigProvider.resolveDebugConfiguration(folder, c))!!
+            const expected: SamLaunchRequestArgs = {
+                type: 'node', // Input "aws-sam", output "node".
+                request: 'attach', // Input "direct-invoke", output "attach".
+                runtime: 'nodejs12.x',
+                runtimeFamily: 1,
+                workspaceFolder: {
+                    index: 0,
+                    name: 'test-workspace-folder',
+                    uri: vscode.Uri.parse(appDir),
+                },
+                baseBuildDir: actual.baseBuildDir, // Random, sanity-checked by assertEqualLaunchConfigs().
+                codeRoot: path.join(appDir, 'src'), // Normalized to absolute path.
+                debugPort: 5858,
+                documentUri: vscode.Uri.parse(''), // TODO: remove or test.
+                handlerName: 'my.test.handler',
+                invokeTarget: {
+                    lambdaHandler: 'my.test.handler',
+                    projectRoot: 'src',
+                    target: 'code',
+                },
+                lambda: {
+                    runtime: 'nodejs12.x',
+                },
+                localRoot: path.join(appDir, 'src'), // Normalized to absolute path.
+                name: 'SamLocalDebug', // "name": "whats in a name"
```

Comment:
Mysterious comment

Reply:
fixed. note that this PR is transitional, there is  low-hanging fruit that will be addressed later.